src/Polly.SharedSpecs/NoOp/NoOpTResultAsyncSpecs.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Polly.NoOp;
+using Polly.Specs.Helpers;
+using Polly.Utilities;
+using Xunit;
+
+namespace Polly.Specs.NoOp
+{
+    public class NoOpTResultAsyncSpecs
+    {
+        [Fact]
+        public void Should_not_throw_when_executing()
+        {
+            NoOpPolicy<int?> policy = Policy.NoOpAsync<int?>();
+            int? result = null;
+
+            policy.Awaiting(async p => result = await p.ExecuteAsync(() => Task.FromResult((int?)10)))
+                .ShouldNotThrow();
+
+            result.HasValue.Should().BeTrue();
+            result.Should().Be(10);
+        }
+
+        [Fact]
+        public void Should_execute_if_cancellationtoken_cancelled_before_delegate_reached()
```

Comment:
Just noticed I forgot to update the name on this spec. It should _not_ cancel! @reisenberger